### PR TITLE
Batch backports to 5.0.x

### PR DESCRIPTION
**Pull request by jlucovsky**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- [4819](https://redmine.openinfosecfoundation.org/issues/4819)
- [4827](https://redmine.openinfosecfoundation.org/issues/4827)
- [4466](https://redmine.openinfosecfoundation.org/issues/4466)
- [4838](https://redmine.openinfosecfoundation.org/issues/4838)
- [4895](https://redmine.openinfosecfoundation.org/issues/4895)
- [4955](https://redmine.openinfosecfoundation.org/issues/4955)
- [4959](https://redmine.openinfosecfoundation.org/issues/4959)
- [4961](https://redmine.openinfosecfoundation.org/issues/4961)
- [4823](https://redmine.openinfosecfoundation.org/issues/4823)
- [4897](https://redmine.openinfosecfoundation.org/issues/4897)
- [4934](https://redmine.openinfosecfoundation.org/issues/4934)
- [4932](https://redmine.openinfosecfoundation.org/issues/4932)

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -294,6 +294,83 @@ jobs:
...
+        run: tar xf prep/suricata-verify.tar.gz
+      - name: Running suricata-verify
+        run: python3 ./suricata-verify/run.py -q
+      # Now install and make sure headers and libraries aren't install
+      # until requested.
+      - run: make install
+      - run: test ! -e /usr/local/lib/libsuricata_c.a
+      - run: test ! -e /usr/local/include/suricata
+
   fedora-32:
```
Comment (jasonish) [inline .github/workflows/builds.yml:374]: Can you also remove any fedora-32 and fedora-33 builds as part of adding fedora 35?  They are both EOL and could randomly start to fail if the package repos get pulled, archived, etc.

Comment (jlucovsky) [reply .github/workflows/builds.yml:374]: Yes -- should fedora-{32,33} be removed for 6.0.x too?

File: src/stream-tcp-reassemble.c [hunk 2]
```diff
@@ -841,9 +841,41 @@ static uint64_t GetStreamSize(TcpStream *stream)
     if (stream) {
         uint64_t size = 0;
         uint32_t cnt = 0;
+        uint64_t last_ack_abs = GetAbsLastAck(stream);
+        uint64_t last_re = 0;
+
+        SCLogNotice("stream_offset %" PRIu64, stream->sb.stream_offset);
```
Comment (victorjulien) [inline src/stream-tcp-reassemble.c:847]: this was fixed in a later commit. But really I think we can drop this commit, its not in any way critical

**Review by victorjulien — CHANGES_REQUESTED** [review]

**Comment by jlucovsky** [comment]
Continued in #6795

**Pull request closed without merging**